web/src/app/pages/edit-survey/edit-job/edit-job.component.ts

```diff
@@ -53,38 +55,38 @@ export class EditJobComponent {
   loiSelection?: LoiSelectionComponent;
 
   constructor(
-    route: ActivatedRoute,
+    private route: ActivatedRoute,
     private navigationService: NavigationService,
     private dialogService: DialogService,
     private surveyService: SurveyService,
     private taskService: TaskService
   ) {
-    this.navigationService.getSurveyId$().subscribe(surveyId => {
-      if (surveyId) {
-        route.params.subscribe(params => {
+    this.subscription.add(
+      this.navigationService.getSurveyId$().subscribe(surveyId => {
+        if (surveyId) {
           this.surveyId = surveyId;
-          this.jobId = params['id'];
-        });
-      }
-    });
+        }
+      })
+    );
   }
 
   async ngOnInit(): Promise<void> {
-    this.tasks = await firstValueFrom(
-      this.surveyService
-        .getActiveSurvey$()
-        .pipe(filter(survey => survey.id === this.surveyId))
-        .pipe(
-          map(survey =>
-            survey
-              .getJob(this.jobId!)
-              ?.tasks?.toList()
-              .sortBy(task => task.index)
+    this.subscription.add(
+      this.route.params.subscribe(async params => {
+        this.jobId = params['id'];
```

Comment:
Please extract to private function

Reply:
what do you mean? @gino-m

Reply:
Since the logic in the callback (async params =>...) is getting a little complex, I was suggesting moving it into a function which you could call from here.